support/vm/vm.go

```diff
@@ -64,6 +66,50 @@ type InternalMessage struct {
 	params interface{}
 }
 
+type ChainMessage struct {
+	Version uint64
+
+	From address.Address
+	To   address.Address
+
+	Nonce uint64
+
+	Value abi.TokenAmount
+
+	GasLimit   int64
+	GasFeeCap  abi.TokenAmount
+	GasPremium abi.TokenAmount
+
+	Method abi.MethodNum
+	Params []byte
+}
+
+func makeChainMessage(from, to address.Address, nonce uint64, value abi.TokenAmount, method abi.MethodNum, params interface{}) (*ChainMessage, error) {
+	var buf bytes.Buffer
+	if params == nil {
+		if err := abi.Empty.MarshalCBOR(&buf); err != nil {
+			return nil, err
+		}
+	} else {
+		if err := params.(cbor.Er).MarshalCBOR(&buf); err != nil {
+			return nil, err
+		}
+	}
+
+	return &ChainMessage{
+		Version:    0,
+		From:       from,
+		To:         to,
+		Nonce:      nonce,
+		Value:      value,
+		GasLimit:   5_000_000_000,
```

Comment:
Reference the default gas limit constant, or use a different value (since I don't think this value is actually checked).